From f222d9c31d7b05fb5302c226285a98337613dbe1 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Thu, 2 Oct 2025 13:51:35 +0200
Subject: [PATCH 275/553] [docs] Prefer Qt::StringLiterals over
 Qt::Literals::StringLiterals

The latter is needlessly verbose. The nested inline namespaces (in Qt
and std alike) are designed to allow skipping the ::literals
middle-man, so let the docs reflect that.

Amends e440fec7fc729342dce7c8421618a95dab11a36b.

Pick-to: 6.8 6.5
Change-Id: I6180833647c8a97a272330663f6e5011a196fcb6
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
Reviewed-by: Paul Wicking <paul.wicking@qt.io>
(cherry picked from commit 9aa0bd9cba720cd64835e8d60d3740ccd2a0b950)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/text/qtliterals.qdoc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/corelib/text/qtliterals.qdoc b/src/corelib/text/qtliterals.qdoc
index eed98eaacc1..c4671415ee4 100644
--- a/src/corelib/text/qtliterals.qdoc
+++ b/src/corelib/text/qtliterals.qdoc
@@ -31,6 +31,8 @@
 
     \code
     // Makes visible only the literal operators declared in StringLiterals
+    using namespace Qt::StringLiterals;
+    // same, but needlessly verbose:
     using namespace Qt::Literals::StringLiterals;
 
     // Makes visible literal operators declared in all inline namespaces
-- 
2.51.2

